SupportFiles/ws-tools.sh

```diff
@@ -0,0 +1,161 @@
+yum -y install git-gui || err_exit "Failed to install git-gui"
+
+# Install ruby
+yum -y install ruby || err_exit "Failed to install ruby"
+
+# Install node.js
+tar -xvf  /etc/cfn/tools/nodejs/node-v11.6.0-linux-x64.tar.xz -C /opt
+chmod -R 755 /opt/node-v11.6.0-linux-x64
+ln -s /opt/node-v11.6.0-linux-x64/bin/node /usr/bin/node
+ln -s /opt/node-v11.6.0-linux-x64/bin/npm /usr/bin/npm
+
+# Install pycharm
+tar -xvf  /etc/cfn/tools/pycharm/pycharm-community-2018.3.3.tar.gz -C /opt
+chmod -R 755 /opt/pycharm-community-2018.3.3
+ln -s /opt/pycharm-community-2018.3.3/bin/pycharm.sh /usr/local/bin/pycharm 
+
+cp /etc/cfn/tools/pycharm/pycharm.desktop /usr/share/applications/pycharm.desktop
+cp /etc/cfn/tools/pycharm/pycharm.desktop /home/WorkstationUser/.local/share/applications/pycharm.desktop
+chown WorkstationUser:WorkstationUser /home/WorkstationUser/.local/share/applications/pycharm.desktop
+chmod 600 /home/WorkstationUser/.local/share/applications/pycharm.desktop     
+
+# Install asciidoctor tool chains 
+rpm -Uvh /etc/cfn/tools/asciidoctor/rubygem-asciidoctor-1.5.6.1-1.el7.noarch.rpm  
+yum -y install rubygem-asciidoctor || err_exit "Failed to install rubygem-asciidoctor"
+
+# Install Visual Studio Code
+rpm -Uvh /etc/cfn/tools/vscode/code-1.30.2-1546901769.el7.x86_64.rpm  
+yum -y install code || err_exit "Failed to install code"       
+
+# Install Mongo db Client – Mongo Shell                                          
+rpm -Uvh /etc/cfn/tools/mongo/mongodb-org-shell-4.0.5-1.el7.x86_64.rpm  
+sudo yum install -y mongodb-org-shell || err_exit "Failed to install mongodb-org-shell"       
+
+# Install MySQL and MySQL Workbench
+# The "proj" libary is required by mysql-workbench but missing from the epel-release-7-11.
+# Therefore, remove epel-release-7-11; and install epel-release-6-8.  We will remove epel-release-6-8
+# and re-install epel-release-7-11 later
+rpm -e epel-release-7-11.noarch
+rpm -Uvh /etc/cfn/tools/mysql/epel-release-6-8.noarch.rpm
+yum -y install proj
+
+rpm -Uvh /etc/cfn/tools/mysql/mysql80-community-release-el7-1.noarch.rpm 
+yum -y install mysql-server || err_exit "Failed to install mysql-server"
+yum -y install mysql-workbench-community || err_exit "Failed to install mysql-workbench"
+
+rpm -Uvh /etc/cfn/tools/mysql/epel-release-7-11.noarch.rpm
+
+systemctl start mysqld
+systemctl enable mysqld
+
+
+# Install Joomla  (Need LAMP stack which includes Apache (2.x+), PHP (5.3.10+)  and MySQL / MariaDB (5.1+)) 
+yum -y install httpd || err_exit "Failed to install httpd"  
+systemctl start httpd
+systemctl enable httpd
+
+yum -y install php php-mysql php-pdo php-gd php-mbstring
+tar -xvf /etc/cfn/tools/joomla/Joomla_3.9.2-Stable-Full_Package.tar.gz -C /var/www/html
+chown -R apache:apache /var/www/html/
+chmod -R 775 /var/www/html/
+systemctl restart httpd
+
+setenforce 0
+#firewall-cmd --permanent --add-service=http
+firewall-cmd --permanent --add-service=https
+firewall-cmd --reload
+setenforce 1
```

Comment:
See prior firewalld-related comment ...and collapse _all_ of your firewall exceptions into one block of code.

Reply:
Not done yet. I didn't see firewall-related comment.